src/data/number.rs

```diff
@@ -445,7 +445,7 @@ impl FromStr for MaybeNumber {
     type Err = LiteralParseError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        Ok(if s.contains(&['p'][..]) {
+        Ok(if s.contains(&['p'][..]) || s.contains('.') {
```

Comment:
I think we can do just 
```suggestion
        Ok(if s.contains('p') || s.contains('.') {
```

Reply:
done